### PR TITLE
Change the DiffuseProbeGrid component to an OBB

### PR DESCRIPTION
**Pull request by dmcdiarmid-ly**

### CODE REVIEW

File: Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp [hunk 1]
```diff
@@ -760,14 +763,15 @@ namespace AZ
...
 
             constantIndex = srgLayout->FindShaderInputConstantIndex(Name("m_modelToWorld"));
-            AZ::Matrix3x4 modelToWorld = AZ::Matrix3x4::CreateFromMatrix3x3AndTranslation(Matrix3x3::CreateIdentity(), m_position) * AZ::Matrix3x4::CreateScale(m_extents);
+            AZ::Matrix3x4 modelToWorld = AZ::Matrix3x4::CreateFromTransform(m_transform) * AZ::Matrix3x4::CreateScale(m_extents);
             m_renderObjectSrg->SetConstant(constantIndex, modelToWorld);
 
-            constantIndex = srgLayout->FindShaderInputConstantIndex(Name("m_aabbMin"));
-            m_renderObjectSrg->SetConstant(constantIndex, m_aabbWs.GetMin());
+            constantIndex = srgLayout->FindShaderInputConstantIndex(Name("m_modelToWorldInverse"));
+            AZ::Matrix3x4 modelToWorldInverse = AZ::Matrix3x4::CreateFromTransform(m_transform).GetInverseFull();
```
Comment (mrieggeramzn) [inline Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp:770]: I'm thinking a full matrix inverse might be slower than calling Transform.GetInverse() and then doing the CreateFromTransform

File: Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp [hunk 2]
```diff
@@ -821,13 +825,14 @@ namespace AZ
...
+            Aabb aabbWs = Aabb::CreateFromObb(m_obbWs);
             Vector3 center;
             float radius;
-            m_aabbWs.GetAsSphere(center, radius);
+            aabbWs.GetAsSphere(center, radius);
 
             m_cullable.m_cullData.m_boundingSphere = Sphere(center, radius);
-            m_cullable.m_cullData.m_boundingObb = m_aabbWs.GetTransformedObb(AZ::Transform::CreateIdentity());
-            m_cullable.m_cullData.m_visibilityEntry.m_boundingVolume = m_aabbWs;
+            m_cullable.m_cullData.m_boundingObb = aabbWs.GetTransformedObb(AZ::Transform::CreateIdentity());
```
Comment (mrieggeramzn) [inline Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp:834]: Is it not possible to pass in m_obbWs?

**Review by mrieggeramzn — COMMENTED** [review]
Not too important right now, but I remember talking with Adi about how creating an AZ::Name is surprisingly expensive. (mutex lock will happen) Maybe cache them in the class?

File: Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp (hunk 2 above)
Comment (dmcdiarmid-ly) [reply Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp]: Done.

File: Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp (hunk 1 above)
Comment (dmcdiarmid-ly) [reply Gems/Atom/Feature/Common/Code/Source/DiffuseGlobalIllumination/DiffuseProbeGrid.cpp:770]: The update currently only happens in the Editor when the user changes the DiffuseProbeGrid in some way, so it's a fairly infrequent occurrence, but if we start moving them around during gameplay I can look at optimizing it.

**Review by mrieggeramzn — APPROVED** [review]

**Review by wintermute-motherbrain — APPROVED** [review]

**Pull request merged**